File: src/broker-service/BrokerService/src/Entities/Accounts/Account.cs
namespace EasyTrade.BrokerService.Entities.Accounts;

public class Account(
    string id,
    string packageId,
    string firstName,
    string lastName,
    string username,
    string email,
    string hashedPassword,
    string origin,
    DateTimeOffset creationDate,
    DateTimeOffset packageActivationDate,
    bool accountActive,
    string address
)
{
    public string Id { get; set; } = id;
    public string PackageId { get; set; } = packageId;
    public string FirstName { get; set; } = firstName;
    public string LastName { get; set; } = lastName;
    public string Username { get; set; } = username;
    public string Email { get; set; } = email;
    public string HashedPassword { get; set; } = hashedPassword;
    public string Origin { get; set; } = origin;
    public DateTimeOffset CreationDate { get; set; } = creationDate;
    public DateTimeOffset PackageActivationDate { get; set; } = packageActivationDate;
    public bool AccountActive { get; set; } = accountActive;
    public string Address { get; set; } = address;
}

File: src/accountservice/src/main/java/com/dynatrace/easytrade/accountservice/Account.java
package com.dynatrace.easytrade.accountservice;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Date;

@Getter
@Setter
@NoArgsConstructor
public class Account {
    private Integer id;
    private Integer packageId;
    private String firstName;
    private String lastName;
    private String username;
    private String email;
    private String hashedPassword;
    private String origin;
    private Date creationDate;
    private Date packageActivationDate;
    private boolean accountActive;
    private String address;

    public ShortAccount toShortAccount() {
        return new ShortAccount(id, username, firstName, lastName);
    }

    public Boolean isPreset() {
        return origin.equals("PRESET");
    }
}
